pom.xml

```diff
@@ -331,36 +355,51 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <version>${maven-surefire-plugin.version}</version>
                 <configuration>
                     <argLine>@{argLine} -Djava.awt.headless=true</argLine>
+                    <excludedGroups>io.confluent.common.utils.IntegrationTest</excludedGroups>
                 </configuration>
                 <executions>
                     <execution>
                         <id>default-test</id>
                         <goals>
                             <goal>test</goal>
                         </goals>
-                        <configuration>
-                            <excludedGroups>io.confluent.common.utils.IntegrationTest</excludedGroups>
-                        </configuration>
                     </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-failsafe-plugin</artifactId>
+                <configuration>
+                    <groups>io.confluent.common.utils.IntegrationTest</groups>
+                    <!-- by default failsafe only includes class names that start or end in IT,
+                         we use regular test class names with @Category and the group above -->
+                    <includes>
+                        <include>**/Test*.java</include>
```

Comment:
Is there any reason not to just use `**/*.java`? The naming restriction seems like it's just a recipe for accidentally not running a test and we already filter by the annotation.

Reply:
I just copied the defaults used by `maven-surefire-plugin` to keep things consistent.

Reply:
Oh, that makes sense. Now I am wondering if we have any unit tests that are not being executed... I sort of assumed anything with the `@Test` annotation would be executed, guess I need to actually learn the java ecosystem's tools properly at some point.